File: GrumpyMapsAngular/src/app/domain/square.ts
import { Player } from './player';

export class Square {
    id:number;
  mapSquareId: number;
  mapCoordinate:string;
  squareHeightWidth: string;
  squareSize: number;
  currentDistance:number = 9999;
  mapHeightWidth: number;
  obstructed: boolean = false;
  inRange: boolean = false;
  mapId:number;
  players: Player[] = new Array();
  numberofPlayers:number;
  zones:any[] = new Array();
  hidden:boolean=false;


  constructor(id:number, mapSquareId: number, squareSize: number, squareHeightWidth: string, heightWidth: number) {
    this.id = id;
    this.mapSquareId = mapSquareId;
    this.squareHeightWidth = squareHeightWidth;
    this.squareSize = squareSize;
    this.mapHeightWidth = heightWidth;
    //this.addTestPlayer();
  }

  setMapId(mapId:number){
      this.mapId = mapId;
  }

  addPhysical(player: Player) {
    player.setMapSquareId(this.mapSquareId);
    player.squareMapCoordinate = this.mapCoordinate;
    this.players.push(player);
    this.numberofPlayers=this.players.length;
  }

  reduceDurations(){
      // ZONES
      var toRemove:any[] = new Array();
      for (var i = 0 ; i < this.zones.length ; i++){
          if (this.zones[i].duration>0){
              this.zones[i].duration--; // reduce duration
          }
          if (this.zones[i].duration==0){ //add zone to remove if duration is 0
              toRemove.push(this.zones[i]);
          }
      }
      for (var i = 0 ; i < toRemove.length ; i++){ // remove all selected zones
          this.removeZone(toRemove[i]);
      }
  }

  removeZone(zone){
      var index = this.zones.indexOf(zone);
      this.zones.splice(index, 1);
  }

  addTestPlayer(){
      if (this.mapSquareId%2==0){
          var player:Player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
      }
      else if (this.mapSquareId%3==0){
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
      }

      else {
          player = new Player(1, 1, "name", 100, 10, 15, 3, 2, "physical", "green", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);
          player= new Player(1, 1, "haha", 100, 10, 15, 3, 2, "physical", "purple", this.mapSquareId, this.mapHeightWidth, this.mapCoordinate, "", this.mapId);
          this.addPhysical(player);

      }
  }

  removePhysical(id) {
    for (var i = 0; i < this.players.length; i++) {
      if (this.players[i].id == id) {
        this.players.splice(i, 1);
        this.numberofPlayers=this.players.length;
      }
    }
  }

}

File: GrumpyMaps/src/main/java/com/grumpymaps/GrumpyMaps/model/Square.java
package com.grumpymaps.GrumpyMaps.model;

import java.io.Serializable;
import java.util.ArrayList;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Transient;

@Entity
public class Square implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private int mapSquareId;
    private String mapCoordinate;
    private String squareHeightWidth;
    private int squareSize;
    private boolean obstructed;
    private int mapHeightWidth;
    private boolean inRange;
    private boolean hidden;


    private Integer mapId;

    @Transient
    private ArrayList<Player> players;
    private ArrayList<Zone> zones;
    private int numberofPlayers;


	/**
	* Returns value of id
	* @return
	*/
	public long getId() {
		return id;
	}

	/**
	* Sets new value of id
	* @param
	*/
	public void setId(long id) {
		this.id = id;
	}

	/**
	* Returns value of mapSquareId
	* @return
	*/
	public int getMapSquareId() {
		return mapSquareId;
	}

	/**
	* Sets new value of mapSquareId
	* @param
	*/
	public void setMapSquareId(int mapSquareId) {
		this.mapSquareId = mapSquareId;
	}

	/**
	* Returns value of mapCoordinate
	* @return
	*/
	public String getMapCoordinate() {
		return mapCoordinate;
	}

	/**
	* Sets new value of mapCoordinate
	* @param
	*/
	public void setMapCoordinate(String mapCoordinate) {
		this.mapCoordinate = mapCoordinate;
	}

	/**
	* Returns value of squareHeightWidth
	* @return
	*/
	public String getSquareHeightWidth() {
		return squareHeightWidth;
	}

	/**
	* Sets new value of squareHeightWidth
	* @param
	*/
	public void setSquareHeightWidth(String squareHeightWidth) {
		this.squareHeightWidth = squareHeightWidth;
	}

	/**
	* Returns value of squareSize
	* @return
	*/
	public int getSquareSize() {
		return squareSize;
	}

	/**
	* Sets new value of squareSize
	* @param
	*/
	public void setSquareSize(int squareSize) {
		this.squareSize = squareSize;
	}

	/**
	* Returns value of obstructed
	* @return
	*/
	public boolean isObstructed() {
		return obstructed;
	}

	/**
	* Sets new value of obstructed
	* @param
	*/
	public void setObstructed(boolean obstructed) {
		this.obstructed = obstructed;
	}

	/**
	* Returns value of mapHeightWidth
	* @return
	*/
	public int getMapHeightWidth() {
		return mapHeightWidth;
	}

	/**
	* Sets new value of mapHeightWidth
	* @param
	*/
	public void setMapHeightWidth(int mapHeightWidth) {
		this.mapHeightWidth = mapHeightWidth;
	}

	/**
	* Returns value of inRange
	* @return
	*/
	public boolean isInRange() {
		return inRange;
	}

	/**
	* Sets new value of inRange
	* @param
	*/
	public void setInRange(boolean inRange) {
		this.inRange = inRange;
	}

	/**
	* Returns value of hidden
	* @return
	*/
	public boolean isHidden() {
		return hidden;
	}

	/**
	* Sets new value of hidden
	* @param
	*/
	public void setHidden(boolean hidden) {
		this.hidden = hidden;
	}

	/**
	* Returns value of players
	* @return
	*/
	public ArrayList<Player> getPlayers() {
		return players;
	}

	/**
	* Sets new value of players
	* @param
	*/
	public void setPlayers(ArrayList<Player> players) {
		this.players = players;
	}

	/**
	* Returns value of dndMap
	* @return
	*/
	public int getMapId() {
		return mapId;
	}

	/**
	* Sets new value of dndMap
	* @param
	*/
	public void setMapId(int mapId) {
		this.mapId = mapId;
	}

    /**
	* Returns value of numberofPlayers
	* @return
	*/
	public int getNumberofPlayers() {
		return numberofPlayers;
	}

	/**
	* Sets new value of numberofPlayers
	* @param
	*/
	public void setNumberofPlayers(int numberofPlayers) {
		this.numberofPlayers = numberofPlayers;
	}

}
